File: service/src/main/java/ch/no1hardy/service/front/dashboard/DashboardData.java
package ch.no1hardy.service.front.dashboard;

import ch.no1hardy.service.front.game.BetGameRes;
import ch.no1hardy.service.front.leaderboard.RankingRes;

import java.util.List;

public record DashboardData(
        List<RankingRes> leaderboardPreview,
        UserSummary userSummary,
        Statistics stats,
        GlobalStatistics globalStats,
        List<BetGameRes> upcomingGames,
        List<BetGameRes> recentResults
) {
}

File: frontend/src/app/model/dashboard/dashboard-data.ts
import {Ranking} from '../leaderboard/ranking'
import {BetGame} from '../game/bet-game'
import {UserSummary} from './user-summary'
import {GlobalStatistics} from './global-statistics'
import {Statistics} from './statistics'

export interface DashboardData {
  leaderboardPreview: (Ranking | null)[]
  userSummary: UserSummary
  stats: Statistics
  globalStats: GlobalStatistics
  openBets: BetGame[]
  recentResults: BetGame[]
}
